### PR TITLE
fix: don't send request reminder for unpaid bookings

### LINKED ISSUE #10253: "Awaiting approval" emails sent for unpaid bookings
### Issue Summary

When a booking is not completed due to the booker abandoning the flow before paying, an email saying "**An event is still waiting for your approval**" is sent to the organizer despite the booking not actually being subject to approval.

### Steps to Reproduce

1. As user A: make a paid event with no requirement for confirmation
2. As user B, fill out the form in booking flow until payment screen
3. As user B, abandon flow and do not paid, resulting in `pending payment` booking
4. As user A, verify the booking shows in Bookings tab and is unconfirmed due to lack of payment
5. Wait for a while (unknown time)

### Actual Results

- User A receives an email unknown amount of time later saying **An event is still waiting for your approval**, making it seem like action is needed from User A

### Expected Results

- Booking should pass without being confirmed
- User A should not be alerted to the booking since it is unconfirmed

### Technical details

- n/a.

### Evidence

Screenshot_20230719-182909

### PR DESCRIPTION
**Pull request by nicktrn**
## What does this PR do?

This prevents "Awaiting Approval" emails from wrongly being sent out for unpaid bookings.

Fixes #10253

It might make sense to create a new "Unpaid Booking" reminder email to send out to Organizers so they can take action, e.g. reject/cancel unpaid bookings. But given the proposed changes to payment flow in #6809, this may be premature.

## Requirement/Documentation

N/A

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

Create 4 bookings:

- A. paid event -> don't pay
- B. paid event with opt-in -> pay
- C. paid event with opt-in -> don't pay
- D. free event with opt-in

Reduce create dates in DB by 3h, then trigger reminder job via:

```sh
APP_URL=localhost:3000
CRON_API_KEY=0cc0e6c35519bba620c9360cfe3e68d0

curl "${APP_URL}/api/cron/bookingReminder" \
  -X POST \
  -H "content-type: application/json" \
  -H "authorization: ${CRON_API_KEY}"
```
See that reminders are only sent for bookings B and D.

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: apps/web/pages/api/cron/bookingReminder.ts [hunk 1]
```diff
@@ -31,6 +31,17 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
         },
         // Only send reminders if the event hasn't finished
         endTime: { gte: new Date() },
+        OR: [
+          // no payment required
+          {
+            payment: { none: {} },
```
Comment (nicktrn) [inline apps/web/pages/api/cron/bookingReminder.ts:37]: this assumes that all bookings that require payment will have one attached

File: apps/web/pages/api/cron/bookingReminder.ts [hunk 2]
```diff
@@ -31,6 +31,17 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
...
+          },
+          // paid but awaiting approval
+          {
+            payment: { some: {} },
+            paid: true,
```
Comment (nicktrn) [inline apps/web/pages/api/cron/bookingReminder.ts:42]: this assumes that no paid bookings can exist where no payment is attached

**Review by nicktrn — COMMENTED** [review]
Ready for review

**Review by Udit-takkar — COMMENTED** [review]
@nicktrn awesome work. changes looks fine to me.

**Review by joeauyeung — APPROVED** [review]
LGTM. Thanks for your contribution @nicktrn

**Pull request merged**